packages/network-controller/tests/NetworkController.test.ts

```diff
@@ -3209,1915 +3468,2439 @@ describe('NetworkController', () => {
   });
 
   describe('upsertNetworkConfiguration', () => {
```

Comment:
I ended up rewriting the test for `upsertNetworkConfiguration`. I hesitated doing this at first because there are a lot of tests here, but I found some duplication and coverage issues with them.

Reply:
I'm going to extract this to make this PR smaller.

Reply:
Extracted here: https://github.com/MetaMask/core/pull/1454